networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -167,7 +173,13 @@ def bfs_edges(G, source, reverse=False, depth_limit=None, sort_neighbors=None):
         successors = G.predecessors
     else:
         successors = G.neighbors
-    yield from generic_bfs_edges(G, source, successors, depth_limit, sort_neighbors)
+    if callable(sort_neighbors):
+        neighbors = sort_neighbors
+    else:
+        neighbors = lambda x: x
+    yield from generic_bfs_edges(
+        G, source, lambda node: iter(neighbors(successors(node))), depth_limit
+    )
```

Comment:
It might be better to put the `yield from` prts of the code into the `if/else` clause. Better in the sense of removing the "identity function" call, and once the deprecation cycle is done, removing the code will require only deletion.
```suggestion
    if callable(sort_neighbors):
        yield from generic_bfs_edges(
            G, source, lambda node: iter(sort_neighbors(successors(node))), depth_limit
        )
    else:
        yield from generic_bfs_edges(G, source, successors, depth_limit)
```